controllers/managedocs_controller.go

```diff
@@ -1752,6 +1762,46 @@ func (r *ManagedOCSReconciler) updateMCGCSV(csv *opv1a1.ClusterServiceVersion) e
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileConsolePlugin() error {
+	r.Log.Info("Reconciling ConsolePlugin")
+	if err := r.get(r.ocpConsolePlugin); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if r.ocpConsolePlugin.UID == "" {
+		// ODF Operator has to bring this CR,
+		// so if OCP ConsolePlugin is not ready, we ruturn silently
+		return nil
+	}
+	expectedProxy := templates.ConsolePluginProxyTemplate.DeepCopy()
+	expectedProxy.Service.Name = prometheusServiceName
+	expectedProxy.Service.Namespace = r.namespace
+
+	var matchingProxy *consolev1alpha1.ConsolePluginProxy
+	for i := range r.ocpConsolePlugin.Spec.Proxy {
+		currProxy := &r.ocpConsolePlugin.Spec.Proxy[i]
+		if currProxy.Service.Name == expectedProxy.Service.Name &&
+			currProxy.Service.Namespace == expectedProxy.Service.Namespace {
+			// if both, expected and current, are same then return happily
+			if reflect.DeepEqual(*currProxy, *expectedProxy) {
+				return nil
+			}
+			matchingProxy = currProxy
+			break
+		}
+	}
+	// if we didn't find any matching ConsolePluginProxy, append the expected proxy
+	if matchingProxy == nil {
+		r.ocpConsolePlugin.Spec.Proxy = append(r.ocpConsolePlugin.Spec.Proxy, *expectedProxy)
+	} else { // else update with the expected proxy
+		expectedProxy.DeepCopyInto(matchingProxy)
+	}
+	// since the proxy array is changed at this point, update the console plugin
+	if err := r.update(r.ocpConsolePlugin); err != nil {
+		return fmt.Errorf("could not update OCP ConsolePlugin: %v", err)
+	}
+	return nil
```

Comment:
This section should handle the 3 cases:
1. no proxy in the original array
2. A proxy in the original array which is configured correctly
3. A proxy in the original array which is not configured correctly

Also, please locate the comments inside the case as they are referring to the case content
```suggestion
	needsUpdate := false
	if matchingProxy == nil {
		// We didn't find any matching ConsolePluginProxy - appending the expected proxy
		r.ocpConsolePlugin.Spec.Proxy = append(r.ocpConsolePlugin.Spec.Proxy, *expectedProxy)
		needsUpdate = true
	} else if !reflect.DeepEqual(*currProxy, *expectedProxy){ 
		// Proxy config do not match - updating with the existing proxy configuration
		expectedProxy.DeepCopyInto(matchingProxy)
		needsUpdate = true		
	}
	if  needsUpdate {
		// Proxy array  changed - updating the console plugin resource
		if err := r.update(r.ocpConsolePlugin); err != nil {
			return fmt.Errorf("could not update OCP ConsolePlugin: %v", err)
		}
	} 
	return nil
```

Reply:
Done, but slightly different way. PTAL